docs/decisions/0012_clickhouse_dbt.rst

```diff
@@ -0,0 +1,65 @@
+12. Clickhouse vs dbt
+#####################
+
+Status
+******
+
+Accepted
+
+Context
+*******
+
+Most of the use cases for Aspects require complex transformations to be performed on the data stored in
+:ref:`clickhouse` before it can be displayed in :ref:`superset`. :ref:`xapi-concepts` is a good format for communicating
+learning-related events as they happen, but in order to analyse them and draw conclusions for what these events signify
+over time, Aspects needs views of the data that are designed specifically for the queries performed.
+
+:ref:`clickhouse` provides some features for transforming complex data, but creating efficient views and managing schema
+changes over time can be difficult. Plus, some of these features (like named collections) are not available on all
+hosted :ref:`clickhouse` services.
+
+This ADR describes the division of responsibilities in Aspects between the :ref:`clickhouse` database and :ref:`dbt`, a
+data transformation tool.
+
+Decision
+********
+
+**Store in Clickhouse, transform with dbt**
+
+Raw event and event sink data tables should continue to be created using :ref:`clickhouse-migrations`.
+
+Data transformations on this raw data should be made with :ref:`dbt`, including:
+
+* materialized views
+* partitions
+* dictionaries
+* fields extracted from event JSON
+
+Note that while we use dbt to manage these data transformations, the transformations themselves 
```

Comment:
```suggestion
Note that while we use dbt to manage the database schema that performs the transformations, the transformations themselves happen in the ClickHouse process. As data is inserted it is immediately transformed and stored in various query-efficient tables.
```

Reply:
Oops! I forgot to finish that thought :)

Have added your content as a "note", see: https://docsopenedxorg--112.org.readthedocs.build/projects/openedx-aspects/en/112/decisions/0012_clickhouse_dbt.html#decision